Improve accessibility of "let's be email friends" form

fstorr opened the pull request:
- Numerous accessibility improvements (see individual commits for details).
- Refactor form using React hooks

jaronheard commented on src/components/shared/mailchimp/mailchimp.js:
```diff
@@ -1,46 +1,82 @@
+    addToMailchimp(email, fields).then(result => {
+      setResultMessage(result.msg.split('<a')[0])
+      setError(result.result === 'error')
+    })
```
I was testing this and it wasn't pulling the form fields into Mailchimp. Instead of using the FormData, we can use the state of the form, which should be more reliable.

```
  const handleSubmit = e => {
    e.preventDefault()
    const email = EMAIL
    const fields = {
      FNAME: FNAME,
      LNAME: LNAME,
    }
    addToMailchimp(email, fields).then(result => {
      setResultMessage(result.msg.split('<a')[0])
      setError(result.result === 'error')
    })
  }
```

jaronheard replied:
Other than that, it is looking good!

fstorr commented:
Okay, cool. Let’s look at that tonight.

On Wed, Oct 16, 2019 at 08:26, Jaron Heard <notifications@github.com> wrote:

> @jaronheard requested changes on this pull request.
>
> ---------------------------------------------------------------
>
> In [src/components/shared/mailchimp/mailchimp.js](https://github.com/odyssy-automaton/civic-web/pull/27#discussion_r335546862):
>
>>      }
> -    console.log(email);
> -    console.log(fields);
> -    addToMailchimp(email, fields).then((result) => {
> -      this.setState({
> -        resultMessage: result.msg.split('<a')[0],
> -        error: result.result === 'error',
> -      })
> -      this.form.reset();
> -    });
> +    addToMailchimp(email, fields).then(result => {
> +      setResultMessage(result.msg.split('<a')[0])
> +      setError(result.result === 'error')
> +    })
>
> I was testing this and it wasn't pulling the form fields into Mailchimp. Instead of using the FormData, we can use the state of the form, which should be more reliable.
>
> const handleSubmit = e => {
>     e.preventDefault()
>     const email = EMAIL
>     const fields = {
>       FNAME: FNAME,
>       LNAME: LNAME,
>     }
>     addToMailchimp(email, fields).then(result => {
>       setResultMessage(result.msg.split('<a')[0])
>       setError(result.result === 'error')
>     })
>   }
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, [view it on GitHub](https://github.com/odyssy-automaton/civic-web/pull/27?email_source=notifications&email_token=AABJL75LJ6O7WGODYINXJYDQO4XBRA5CNFSM4JAXE6A2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCIFFUUI#pullrequestreview-302668369), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/AABJL76VQ5VV7TE6YW44FIDQO4XBRANCNFSM4JAXE6AQ).
